### PR TITLE
Ready: Implement StyleCop

### PR DESCRIPTION
**Pull request by lukebakken**
To start, I modified the And.cs file to not have any StyleCop warnings based on stock StyleCop settings.

CLIENTS-146 in JURA

### CODE REVIEW

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs [hunk 1]
```diff
@@ -13,64 +15,90 @@
...
-using System.Text;
+// </copyright>
 
 namespace RiakClient.Models.MapReduce.KeyFilters
 {
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using Newtonsoft.Json;
```
Comment (lukebakken) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: Personally, I like having `using` statements inside of the namespace for the reasons outlined here:

http://stackoverflow.com/a/151560

Resharper settings to implement it: http://stackoverflow.com/a/8163511

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs [hunk 2]
```diff
@@ -13,64 +15,90 @@
...
-        public string FunctionName
+        /// <summary>
+        /// Initializes a new instance of the <see cref="And"/> class.
+        /// </summary>
+        /// <param name="left">left hand token</param>
+        /// <param name="right">right hand token</param>
+        public And(List<IRiakKeyFilterToken> left, List<IRiakKeyFilterToken> right)
         {
-            get { return _kfDefinition.Item1; }
+            this.keyFilterDefinition = Tuple.Create("and", left, right);
```
Comment (lukebakken) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: Believe it or not, requiring `this.` is an OOTB rule. I propose disabling that rule.

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs [hunk 3]
```diff
@@ -13,64 +15,90 @@
...
     /// <summary>
     /// Joins two or more key-filter operations with a logical AND operation.
     /// </summary>
     internal class And : IRiakKeyFilterToken
     {
-        private readonly Tuple<string, List<IRiakKeyFilterToken>, List<IRiakKeyFilterToken>> _kfDefinition;
+        /// <summary>
+        ///  Collection of key filter definitions.
+        /// </summary>
+        private readonly Tuple<string, List<IRiakKeyFilterToken>, List<IRiakKeyFilterToken>> keyFilterDefinition;
```
Comment (lukebakken) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: The default rule is to *not* use an underscore prefix, which I do prefer, even if it is different than `corefx` formatting.

Interesting discussion here: http://blogs.msdn.com/b/sourceanalysis/archive/2008/05/25/a-difference-of-style.aspx

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs [hunk 4]
```diff
@@ -1,4 +1,6 @@
+// <copyright file="And.cs" company="Basho Technologies, Inc.">
```
Comment (lukebakken) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: Using `<copyright file="" company="">` is an OOTB rule that I don't mind.

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs [hunk 5]
```diff
@@ -13,64 +15,90 @@
...
+        /// Converts the current token to JSON
+        /// </summary>
+        /// <returns>JSON representation of the <see cref="And"/> class</returns>
         public string ToJsonString()
         {
             var sb = new StringBuilder();
 
-            using(var sw = new StringWriter(sb))
-            using(JsonWriter jw = new JsonTextWriter(sw))
+            using (var sw = new StringWriter(sb))
```
Comment (lukebakken) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: No rule about missing `{ }`. I am :disappointed:

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs [hunk 6]
```diff
@@ -13,64 +15,90 @@
...
+        ///  Collection of key filter definitions.
+        /// </summary>
+        private readonly Tuple<string, List<IRiakKeyFilterToken>, List<IRiakKeyFilterToken>> keyFilterDefinition;
 
-        public string FunctionName
+        /// <summary>
+        /// Initializes a new instance of the <see cref="And"/> class.
+        /// </summary>
+        /// <param name="left">left hand token</param>
+        /// <param name="right">right hand token</param>
```
Comment (lukebakken) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: I suspect if we leave the documentation rules in place at this time, we'll go insane. Maybe we should enable them once we get close to the first official release.

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs (hunk 2 above)
Comment (alexmoore) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: +1 to disabling it.

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs (hunk 4 above)
Comment (alexmoore) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: +1.

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs (hunk 3 above)
Comment (alexmoore) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: I also like avoiding pseudo hungarian notation, I think Uncle Bob warns of it in "Clean Code" too.  Most of the time we're using intellisense so the extra characters aren't an issue.  

I would vote for the CoreFX version of underscore usage over `this` here, it's mostly a personal preference though.  Using a `_` gives the reader another clue to the scoping of the variable, which is nice in places where you don't have "Find Declaration..." (i.e. github reviews).  You can omit a `this.` however if you don't have stylecop error out things in the build.  6 one way, half a dozen the other.

File: src/RiakClient/Models/MapReduce/KeyFilters/And.cs (hunk 6 above)
Comment (alexmoore) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: We should start with public/protected members/properties/methods only.

Comment (lukebakken) [inline src/RiakClient/Models/MapReduce/KeyFilters/And.cs]: And of course you can't change this by visibility :-/

**Comment by lukebakken** [comment]
@alexmoore my main concern with all of these changes are situations where `_varFoo` was renamed to `varFoo` when an existing `varFoo` was already being used, like in this place:

https://github.com/basho-labs/riak-dotnet-client/blob/jira/lrb/clients-146/src/RiakClient/Models/Search/GroupTerm.cs#L34-L41

https://github.com/basho-labs/riak-dotnet-client/blob/jira/lrb/clients-146/src/RiakClient/Models/Search/RiakFluentSearch.cs#L117-L124

**Pull request merged**